lib/engine/game/g_1849/game.rb

```diff
@@ -532,13 +573,25 @@ def check_distance(route, _visits)
         end
 
         def check_other(route)
+          check_track_type(route)
           return unless (route.stops.map(&:hex).map(&:id) & PORT_HEXES).any?
 
           raise GameError, 'Route must include two non-port stops.' unless route.stops.size > 2
         end
 
+        def check_track_type(route)
+          train = route.train
+          paths = route.paths
+
+          raise GameError, 'E-Trains cannot use narrow gauge' if paths.any? { |p| p.track == :narrow } && train.name == 'E'
```

Comment:
For performance, check the train name first, so we can stop checking if not an E train.

Reply:
Good call, switched.